itest/lnd_remote_signer_test.go

```diff
@@ -202,39 +259,191 @@ func prepareRemoteSignerTest(ht *lntest.HarnessTest, tc remoteSignerTestCase) (
+			"--remotesigner.allowinboundconnection",
+			"--remotesigner.timeout=30s",
+			"--remotesigner.requesttimeout=30s",
+		}, commitArgs...),
+		password, true,
+	)
+
+	// As the signer node will make an outbound connection to the
+	// watch-only node, we must specify the watch-only node's RPC
+	// connection details in the signer's configuration.
+	signerArgs := []string{
+		"--watchonlynode.enable",
+		"--watchonlynode.timeout=30s",
+		"--watchonlynode.requesttimeout=10s",
+		fmt.Sprintf(
+			"--watchonlynode.rpchost=localhost:%d",
+			watchOnly.Cfg.RPCPort,
+		),
+		fmt.Sprintf(
+			"--watchonlynode.tlscertpath=%s",
+			watchOnly.Cfg.TLSCertPath,
+		),
+		fmt.Sprintf(
+			"--watchonlynode.macaroonpath=%s",
+			watchOnly.Cfg.AdminMacPath,
+		),
+	}
+
+	if !tc.randomSeed {
+		signer = ht.RestoreNodeWithSeed(
+			"Signer", signerArgs, password, nil, rootKey, 0,
+			nil,
+		)
+	} else {
+		signer = ht.NewNode("Signer", signerArgs)
+		signerNodePubKey = signer.PubKeyStr
+
+		rpcAccts := signer.RPC.ListAccounts(
+			&walletrpc.ListAccountsRequest{},
+		)
+
+		watchOnlyAccounts, err = walletrpc.AccountsToWatchOnly(
+			rpcAccts.Accounts,
+		)
+		require.NoError(ht, err)
+	}
+
+	// As the watch-only node will not fully start until the signer
+	// node connects to it, we need to start the watch-only node
+	// after having started the signer node.
+	ht.StartWatchOnly(watchOnly, "WatchOnly", password,
+		&lnrpc.WatchOnly{
+			MasterKeyBirthdayTimestamp: 0,
+			MasterKeyFingerprint:       nil,
+			Accounts:                   watchOnlyAccounts,
+		},
+	)
+
+	resp := watchOnly.RPC.GetInfo()
+	require.Equal(ht, signerNodePubKey, resp.IdentityPubkey)
+
+	if tc.sendCoins {
+		ht.FundCoins(btcutil.SatoshiPerBitcoin, watchOnly)
+		ht.AssertWalletAccountBalance(
+			watchOnly, "default",
+			btcutil.SatoshiPerBitcoin, 0,
+		)
+	}
+
+	carol := ht.NewNode("carol", commitArgs)
+	ht.EnsureConnected(watchOnly, carol)
+
+	return signer, watchOnly, carol
+}
+
+func executeRemoteSignerTestCase(ht *lntest.HarnessTest,
+	tc remoteSignerTestCase, isOutbound bool) {
+
+	var watchOnly, carol *node.HarnessNode
+
+	if isOutbound {
+		_, watchOnly, carol = prepareOutboundRemoteSignerTest(ht, tc)
+	} else {
+		_, watchOnly, carol = prepareInboundRemoteSignerTest(ht, tc)
+	}
+
+	tc.fn(ht, watchOnly, carol)
+}
+
+func randomSeedTestCase(isOutbound bool) remoteSignerTestCase {
+	tName := "random seed"
+	if isOutbound {
+		tName += " outbound"
+	}
+
+	return remoteSignerTestCase{
+		name:       tName,
 		randomSeed: true,
 		fn: func(tt *lntest.HarnessTest, wo, carol *node.HarnessNode) {
 			// Nothing more to test here.
 		},
 	}
+}
 
-	_, watchOnly, carol := prepareRemoteSignerTest(ht, tc)
-	tc.fn(ht, watchOnly, carol)
+// testRemoteSignerRandomSeed tests that a watch-only wallet can use a remote
+// signing wallet to perform any signing or ECDH operations.
+func testRemoteSignerRandomSeed(ht *lntest.HarnessTest) {
+	executeRemoteSignerTestCase(ht, randomSeedTestCase(false), false)
```

Comment:
Usually the pattern that when two boolean share the same state, it usually means one of the flags is redundant, in this case i think passing this to the test case to change the `tName` is redundant, as we already have defined the names in `remoteSignerTestCases`